adaptive/src/main/java/com/google/accompanist/adaptive/TwoPane.kt

```diff
@@ -0,0 +1,545 @@
+            val detailOffsetY =
+                if (isHorizontalSplit) {
+                    0
+                } else {
+                    constraints.maxHeight - secondPlaceable.height
+                }
+            secondPlaceable.placeRelative(detailOffsetX, detailOffsetY)
+        }
+    }
+}
+
+/**
+ * Returns the specification for where to place a split in [TwoPane] as a result of
+ * [TwoPaneStrategy.calculateSplitResult]
+ */
+public class SplitResult(
+
+    /**
+     * Whether the split is vertical or horizontal
+     */
+    public val isHorizontalSplit: Boolean,
+
+    /**
+     * The bounds that are nether a `start` pane or an `end` pane, but a separation between those
+     * two. In case width or height is 0 - it means that the gap itself is a 0 width/height, but the
+     * place within the layout is still defined.
+     *
+     * The [gapBounds] should be defined in local bounds to the [TwoPane].
+     */
+    public val gapBounds: Rect,
+)
+
+/**
+ * A strategy for configuring the [TwoPane] component, that is responsible for the meta-data
+ * corresponding to the arrangement of the two panes of the layout.
+ */
+public fun interface TwoPaneStrategy {
+    /**
+     * Calculates the split result in local bounds of the [TwoPane].
+     *
+     * @param density the [Density] for measuring and laying out
+     * @param layoutDirection the [LayoutDirection] for measuring and laying out
+     * @param layoutCoordinates the [LayoutCoordinates] of the [TwoPane]
+     */
+    public fun calculateSplitResult(
+        density: Density,
+        layoutDirection: LayoutDirection,
+        layoutCoordinates: LayoutCoordinates
+    ): SplitResult
+}
+
+/**
+ * A strategy for configuring the [TwoPane] component, that is responsible for the meta-data
+ * corresponding to the arrangement of the two panes of the layout.
+ *
+ * This strategy can be conditional: If `null` is returned from [calculateSplitResult], then this
+ * strategy did not produce a split result to use, and a different strategy should be used.
+ */
+private fun interface ConditionalTwoPaneStrategy {
+    /**
+     * Calculates the split result in local bounds of the [TwoPane], or `null` if this strategy
+     * does not apply.
+     *
+     * @param density the [Density] for measuring and laying out
+     * @param layoutDirection the [LayoutDirection] for measuring and laying out
+     * @param layoutCoordinates the [LayoutCoordinates] of the [TwoPane]
+     */
+    public fun calculateSplitResult(
+        density: Density,
+        layoutDirection: LayoutDirection,
+        layoutCoordinates: LayoutCoordinates
+    ): SplitResult?
+}
+
+/**
+ * Returns a [TwoPaneStrategy] that will place the slots vertically or horizontally if there is a
+ * horizontal or vertical fold respectively.
+ *
+ * If there is no fold, then the [defaultStrategy] will be used instead.
+ */
+public fun TwoPaneStrategy(
+    windowGeometry: WindowGeometry,
+    defaultStrategy: TwoPaneStrategy,
```

Comment:
I like the name `default` more. What I was wondering if we need this wrapping at all? Having to create a strategy and provide a default one for the non-fold seems like an unnecessary step for developers. They might (should) use the other ones, but this function take the best name in the space. 

What do you think if we have just remove this from public api?

Reply:
If we remove this one from the public API, then I think we would need to expose how to create the fold strategies to allow for the same behavior, which would also mean exposing the idea of a strategy that may not apply. Not sure if we'll be able to completely avoid having a `default`.

Reply:
In the latest commit, I was able to keep current functionality intact while removing the direct fallback support from the public API. Let me know what you think!